### DISCUSSION TITLE
crc api remote access from mac os doesn't work

### DISCUSSION THREAD

**Discussion by imperialguy** [crc-org/crc#3472] [Q&A] [answered]
On Mac OS Ventura 13.1, I have the following:

```
$ cat /usr/local/etc/dnsmasq.d/development.conf
address=/apps-crc.testing/<openshift_server_ip_addr>
address=/api.crc.testing/<openshift_server_ip_addr>
```

The following works perfectly in my browser --> `https://console-openshift-console.apps-crc.testing`. Meaning, I am able to access the openshift console from Mac OS.

But, the following fails:

```
$ oc login https://api.crc.testing:6443 --insecure-skip-tls-verify=true --token=sha256~xxxxxxxxx
error: dial tcp: lookup api.crc.testing on 8.8.8.8:53: no such host - verify you have provided the correct host and port and that the server is currently running.
```

I tried putting the following line in both `/etc/resolv.conf` and `/etc/resolver/testing` files as well:

```
nameserver <openshift_server_ip_addr>
```

Still doesn't work. Any ideas?

Also asked [here](https://gist.github.com/tmckayus/8e843f90c44ac841d0673434c7de0c6a?permalink_comment_id=4425316#gistcomment-4425316)

**Comment by cfergeau** [comment, marked as answer]
`oc` won't make use of `/etc/resolver/testing`, I'd expect setting the nameserver in `/etc/resolv.conf` to work though. 
Maybe you can add entries to `/etc/hosts` instead?

**Reply by imperialguy** [reply]
Okay, so I added the following line to `/etc/hosts`:

```
<openshift_server_ip_address> api.crc.testing
```

Looks like the tcp dial error is gone. But now I get this error:

```
$ oc login https://api.crc.testing:6443 --insecure-skip-tls-verify=true --token=sha256~xxxx
error: x509: “kube-apiserver-lb-signer” certificate is not trusted
```

Any ideas?

**Reply by imperialguy** [reply]
Nvm..this [hack](https://bugzilla.redhat.com/show_bug.cgi?id=2097830#c27) worked.
